lib/pkcs11_req.c

```diff
@@ -1098,25 +1101,35 @@ int pkcs11_sign_X509_REQ(pkcs11Context * p11Context, CK_VOID_PTR req, int output
     {
 
 	ECDSA_SIG *ecsig = ECDSA_SIG_new();
+	BIGNUM *sig_r;
+	BIGNUM *sig_s;
 
 	int siglen;
 	unsigned char *sigder = NULL;
 
 	if(ecsig==NULL) {
 	    goto err;
 	}
+	if((sig_r = BN_new()) == NULL) {
```

Comment:
Since you are now allocating `sig_r` and `sig_s`, you should free memory in case of error ( `err:` label, around line 1210 ).